hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/spi/impl/KeyPrefixContainerCodec.java

```diff
@@ -50,6 +56,130 @@ public Class<KeyPrefixContainer> getTypeClass() {
     return KeyPrefixContainer.class;
   }
 
+  @Override
+  public boolean supportCodecBuffer() {
+    return true;
+  }
+
+  @Override
+  public CodecBuffer toCodecBuffer(@Nonnull KeyPrefixContainer object, CodecBuffer.Allocator allocator) {
+    Preconditions.checkNotNull(object, "Null object can't be converted to CodecBuffer.");
+
+    final byte[] keyPrefixBytes = object.getKeyPrefix().getBytes(UTF_8);
+    int totalSize = keyPrefixBytes.length;
+
+    if (object.getKeyVersion() != -1) {
+      totalSize += KEY_DELIMITER.getBytes(UTF_8).length + Long.BYTES;
+    }
+    if (object.getContainerId() != -1) {
+      totalSize += KEY_DELIMITER.getBytes(UTF_8).length + Long.BYTES;
+    }
+
+    final CodecBuffer buffer = allocator.apply(totalSize);
+    buffer.put(ByteBuffer.wrap(keyPrefixBytes));
+
+    if (object.getKeyVersion() != -1) {
+      buffer.put(ByteBuffer.wrap(KEY_DELIMITER.getBytes(UTF_8)));
+      buffer.putLong(object.getKeyVersion());
+    }
+
+    if (object.getContainerId() != -1) {
+      buffer.put(ByteBuffer.wrap(KEY_DELIMITER.getBytes(UTF_8)));
+      buffer.putLong(object.getContainerId());
+    }
+
+    return buffer;
+  }
+
+  @Override
+  public KeyPrefixContainer fromCodecBuffer(@Nonnull CodecBuffer buffer) throws CodecException {
+
+    Object wrapped = buffer.getWrapped();
+    if (wrapped instanceof byte[]) {
+      return fromPersistedFormat((byte[]) wrapped);
+    }
```

Comment:
@YutaLin , one more thing: Let's remove this?  The test use heap buffers.  Then, it will have arrays and won't test the code below this.